api/apiclient.go

```diff
@@ -1280,13 +1280,13 @@ func (c *conn) Broken() <-chan struct{} {
 }
 
 // IsBroken implements api.Connection.
-func (c *conn) IsBroken() bool {
+func (c *conn) IsBroken(ctx context.Context) bool {
 	select {
 	case <-c.broken:
 		return true
 	default:
 	}
```

Comment:
We could do a select on the context here, to prevent wasting CPU cycles doing the ping. Not 100% sure what the return value should be. It feels like `IsBroken` should return an error.

```suggestion
	select {
	case <-c.broken:
		return true
	case <-ctx.Done():
		return ???	
	default:
	}
```

Reply:
I think for now return true if context is done is ok. That's what it will return anyway if the context expires during the ping.